src/InvertedIndices.jl

```diff
@@ -160,4 +164,9 @@ end
 @inline Base.to_indices(A, I::Tuple{Not{<:AbstractArray{Bool, 1}}}) = to_indices(A, (eachindex(IndexLinear(), A),), I)
 @inline Base.to_indices(A, I::Tuple{Not{<:Union{Array{Bool}, BitArray}}}) = to_indices(A, (eachindex(A),), I)
 
+# a cleaner implementation is nt[filter(∉(I.skip), keys(nt))] instead of Base.structdiff, but this would only work on Julia 1.7+
+@inline Base.getindex(nt::NamedTuple, I::InvertedIndex{Symbol}) = I.skip ∈ keys(nt) ? Base.structdiff(nt, NamedTuple{(I.skip,)}) : error("type NamedTuple has no field $(I.skip)")
+@inline Base.getindex(nt::NamedTuple, I::InvertedIndex{NTuple{N, Symbol}}) where {N} = I.skip ⊆ keys(nt) ? Base.structdiff(nt, NamedTuple{I.skip}) : error("type NamedTuple has no fields $(join(I.skip, ", "))")
+@inline Base.getindex(nt::NamedTuple, I::InvertedIndex{<:AbstractVector{Symbol}}) = getindex(nt, Not(Tuple(I.skip)))
```

Comment:
1. can you please refactor the code to make lines be narrower (preferably below 92 characters)
2. Instead of `error` we should `throw` a more specific error type here I think. I know that in `NamedTuple` implementation `ErrorException` is thrown, but it seems not a good choice to me. @nalimilan - do you think `ArgumentError` is better? I am also OK with leaving `ErrorException`.

Reply:
I agree `ErrorException` isn't the best choice, but given that this is what Base throws it makes sense to do the same (until it's changed there).